app/scripts/modules/core/widgets/spelText/jsonListBuilder.spec.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+describe('jsonListBuilder', function () {
+
+  let builder;
+
+  beforeEach(
+    window.module(
+      require('./jsonListBuilder')
+    )
+  );
+
+  beforeEach(
+    window.inject(function(jsonListBuilder) {
+      builder = jsonListBuilder;
+    })
+  );
+
+  describe('create a list with only the leaf keys', function () {
+    it('a simple json with one attribute', function () {
+      let json = { name: 'foo'};
+      let result = builder.convertJsonKeysToBracketedList(json);
+      expect(result).toEqual([`['name']`]);
+    });
+
+
+    it('a simple json with multiple attributes', function () {
+      let json = { name: 'foo', bar: 'baz'};
+      let result = builder.convertJsonKeysToBracketedList(json);
+      expect(result).toEqual([`['name']`, `['bar']`]);
+    });
+
+    it('nested objects', function () {
+      let json = { name: {foo: 'bar'}};
+      let result = builder.convertJsonKeysToBracketedList(json);
+      expect(result).toEqual([`['name']['foo']`]);
+    });
+
+    it('nested objects with multivalues', function () {
+      let json = { name: {foo: 'bar', baz: 2}};
+      let result = builder.convertJsonKeysToBracketedList(json);
+      expect(result).toEqual([`['name']['foo']`, `['name']['baz']`]);
+    });
+
+    it('deep nested objects', function () {
+      let json = { name: { baz: { context: {foo: 2}}}};
+      let result = builder.convertJsonKeysToBracketedList(json);
+      expect(result).toEqual([`['name']['baz']['context']['foo']`]);
+    });
+
+    it('deep nested objects with multivalues', function () {
+      let json = { name: { baz: { context: {foo: 2, boom: 'go'}}}};
+      let result = builder.convertJsonKeysToBracketedList(json);
+      expect(result).toEqual([ `['name']['baz']['context']['foo']`, `['name']['baz']['context']['boom']`]);
+    });
+
+    it('json with an array of strings', function () {
+      let json = { name: ['foo'] };
+      let result = builder.convertJsonKeysToBracketedList(json);
+      expect(result).toEqual([`['name'][0]`]);
+    });
+
+    it('json with an array of multiple strings', function () {
+      let json = { name: ['foo', 'bar'] };
+      let result = builder.convertJsonKeysToBracketedList(json);
+      expect(result).toEqual([`['name'][0]`, `['name'][1]`]);
+    });
+
+    it('json with an array of one object', function () {
+      let json = { name: [{foo: 2}] };
+      let result = builder.convertJsonKeysToBracketedList(json);
+      expect(result).toEqual([`['name'][0]['foo']`]);
+    });
+
+    it('json with an array of two object', function () {
+      let json = { name: [{foo: 2}, {foo:3}] };
+      let result = builder.convertJsonKeysToBracketedList(json);
+      expect(result).toEqual([`['name'][1]['foo']`, `['name'][0]['foo']`]);
+    });
```

Comment:
Might be good to include a test verifying that we're handling keys with dots, e.g. `notification.type` or `deploy.server.groups`